File: LeetCode/1497-check-if-array-pairs-are-divisible-by-k/1497-check-if-array-pairs-are-divisible-by-k.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        brr = [0] * k
        for x in arr:
            rem = ((x % k) + k) % k
            brr[rem] += 1
        
        if brr[0] % 2 != 0:
            return False
        
        for rem in range(1, (k // 2) + 1):
            if brr[rem] != brr[k - rem]:
                return False
        
        return True

File: LeetCode/1497-check-if-array-pairs-are-divisible-by-k/1497-check-if-array-pairs-are-divisible-by-k.java
class Solution {
    public boolean canArrange(int[] arr, int k) {
        int[] brr = new int[k];
        for (int x : arr) {
            int rem = ((x % k) + k) % k;
            brr[rem]++;
        }
        if (brr[0] % 2 == 1) {
            return false;
        }
        for (int rem = 1; rem <= k / 2; rem++) {
            int half = k - rem;
            if (brr[half] != brr[rem]) {
                return false;
            }
        }
        return true;
    }
}